pkg/resourcemanager/azuresql/azuresqlmanageduser/azuresqlmanageduser_reconcile.go

```diff
@@ -146,7 +146,24 @@ func (s *AzureSqlManagedUserManager) Delete(ctx context.Context, obj runtime.Obj
 		requestedUsername = instance.Name
 	}
 
-	db, err := s.ConnectToSqlDbAsCurrentUser(ctx, DriverName, instance.Spec.Server, instance.Spec.DbName)
+	// short circuit connection if database doesn't exist
+	_, err = s.GetDB(ctx, instance.Spec.ResourceGroup, instance.Spec.Server, instance.Spec.DbName)
```

Comment:
Was the underlying problem that the connect call would hang if the database doesn't exist? Should we also add a timeout on the connect? It seems like there could be other problems where the DB exists but connect still hangs - presumably that would block reconciliation.

Reply:
Doesn't hang - errored with connection rejected.

Reply:
Oh ok.

Reply:
It looks like the AutoRest clients already have timeouts configured:
```
	transport := &http.Transport{
		Proxy:                 defaultTransport.Proxy,
		DialContext:           defaultTransport.DialContext,
		MaxIdleConns:          defaultTransport.MaxIdleConns,
		IdleConnTimeout:       defaultTransport.IdleConnTimeout,
		TLSHandshakeTimeout:   defaultTransport.TLSHandshakeTimeout,
		ExpectContinueTimeout: defaultTransport.ExpectContinueTimeout,
		TLSClientConfig: &tls.Config{
			MinVersion:    tls.VersionTLS12,
			Renegotiation: renengotiation,
		},
	}
```

Values seem to be this:
```
	Proxy: ProxyFromEnvironment,
	DialContext: (&net.Dialer{
		Timeout:   30 * time.Second,
		KeepAlive: 30 * time.Second,
		DualStack: true,
	}).DialContext,
	ForceAttemptHTTP2:     true,
	MaxIdleConns:          100,
	IdleConnTimeout:       90 * time.Second,
	TLSHandshakeTimeout:   10 * time.Second,
	ExpectContinueTimeout: 1 * time.Second,
```

So I think we're good there.